dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/config/StoreConfiguration.java

```diff
@@ -43,6 +46,12 @@ public StorageOperate storageOperate() {
         switch (PropertyUtils.getString(RESOURCE_STORAGE_TYPE)) {
             case STORAGE_S3:
                 return S3Utils.getInstance();
+            case STORAGE_LOCAL:
+                // override the FS_DEFAULT_FS configuration
+                PropertyUtils.setValue(FS_DEFAULT_FS,STORAGE_LOCAL_DEFAULT_FS);
```

Comment:
This would be overwrite by
https://github.com/SbloodyS/dolphinscheduler/blob/4c1ef0a89a81e3e7464300a7d010a4786abd3e7b/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java#L125

Reply:
![image](https://user-images.githubusercontent.com/50058173/170659285-49ff7971-c8b9-452f-8876-702e7cdd3897.png)

What you described should be my screenshot. In fact, you can find that the `defaultFSProp `in `configuration.set(Constants.FS_DEFAULT_FS, defaultFSProp); `is the value I put in in advance

Reply:
> String defaultFS = configuration.get(Constants.FS_DEFAULT_FS);

`String defaultFS = configuration.get(Constants.FS_DEFAULT_FS); `
the `defaultFS` is always "file:///" , so ` defaultFS.startsWith("file")` is always true ，and `defaultFSProp `is the value configured in the configuration file, which is the value I passed in 
![image](https://user-images.githubusercontent.com/50058173/170685278-0debe19e-0586-4f54-856d-5547b9198ec1.png)
I output these two values on the code, let's take a look：
![image](https://user-images.githubusercontent.com/50058173/170685541-a940c7f1-c7b7-430a-85b9-92f00e0bdc3a.png)
![image](https://user-images.githubusercontent.com/50058173/170685860-1e995e32-a1dd-4e39-8445-a91992ea6eb8.png)
The logic of my pr can be seen as the following pseudo code:
`if (resource.storage.type==HDFS_LOCAL) fs.defaultFS=file:///`

Reply:
@zhongjiajie PTAL

Reply:
LGTM overall, but set PropertyUtils in StoreConfiguration is a litter odd IMO, If I do this job, I will add some callback func or a hook in https://github.com/apache/dolphinscheduler/blob/996790ce9ea48bd96aed6687d31afe931fc70893/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/PropertyUtils.java#L65 after all property set.